lib/utils/colors.js

```diff
@@ -2,132 +2,120 @@ const Settings = {
-      if (background !== undefined) {
-        string += background();
-      }
+class ConsoleColor {
+  constructor() {
+    this.background = new Background();
 
-      string += `${text}\u{1b}[0m`;
+    Object.keys(foregroundColors).forEach(k => {
+      ConsoleColor.prototype[k.toLowerCase()] = (text, background) => {
+        if (!Settings.enabled) {
+          return text;
+        }
 
-      return string;
-    };
-  });
+        let string = `\u{1b}[${foregroundColors[k.toLowerCase()]}m`;
+        if (background !== undefined) {
+          string += background();
+        }
 
-  Object.keys(backgroundColors).forEach(k => {
-    Background.prototype[k.toLowerCase()] = (text) => {
-      return `\u{1b}[${backgroundColors[k.toLowerCase()]}m`;
-    };
-  });
+        string += `${text}\u{1b}[0m`;
 
-  return this;
+        return string;
+      };
+    });
+
+    Object.keys(backgroundColors).forEach(k => {
+      Background.prototype[k.toLowerCase()] = () => {
+        return `\u{1b}[${backgroundColors[k.toLowerCase()]}m`;
+      };
+    });
+  }
 }
 
-module.exports = new (function() {
-  let instance;
+class Colors {
+  constructor() {
+    this.initialize();
+  }
 
-  const disable = function() {
+  disable() {
     Settings.enabled = false;
 
     Object.keys(ConsoleColor.prototype).forEach(function (color) {
       ConsoleColor.prototype[color] = function (text) {
         return text;
       };
-    });
-  };
+    }); 
+  }
 
-  const enable = function() {
+  enable() {
     Settings.enabled = true;
-  };
+  }
 
-  const setup = function() {
-    instance = new ConsoleColor();
-  };
+  setup() {
+    this.instance = new ConsoleColor();
+  }
 
-  const initialize = function() {
+  initialize() {
     if (!Settings.enabled) {
-      disable();
+      this.disable();
     }
 
-    setup();
-  };
-
-  ////////////////////////////////////////////////////////////
-  // Public methods
-  ////////////////////////////////////////////////////////////
-  const exported = {
-    disableColors() {
-      disable();
-      setup();
-    },
-    disable() {
-      disable();
-      setup();
-    },
-    enable() {
-      enable();
-      setup();
-    }
-  };
+    this.setup();
+  }
+}
 
-  Object.defineProperty(exported, 'colors', {
-    configurable: true,
-    get: function() {
-      return instance;
-    }
-  });
+const colorsObj = new Colors();
 
-  Object.defineProperty(exported, 'colorsEnabled', {
-    configurable: true,
-    get: function() {
-      return Settings.enabled;
-    }
-  });
+const disable = function() {
+  colorsObj.disable();
+  colorsObj.setup();
+};
+const disableColors = disable;
+
+const enable = function() {
+  colorsObj.enable();
+  colorsObj.setup();
+};
 
-  initialize();
+const colors = colorsObj.instance;
+const colorsEnabled = Settings.enabled;
 
-  return exported;
-});
+module.exports = {
+  disable,
+  disableColors,
+  enable,
+  colors,
+  colorsEnabled
+};
```

Comment:
A possible way to refactor the code is by adding the  `ConsoleColors` class and adding`new ConsoleColors()` statement at the end. This will create a new instance of the ConsoleColors class that is similar to the exported object in the original implementation. 


```javascript
class ConsoleColors {
  constructor() {
    this.instance = null;
    this.enabled = process.env.COLORS !== '0';
    this.initialize();
  }

  disable() {
    this.enabled = false;

    Object.keys(ConsoleColor.prototype).forEach(function (color) {
      ConsoleColor.prototype[color] = function (text) {
        return text;
      };
    });
  }

  enable() {
    this.enabled = true;
  }

  initialize() {
    if (!this.enabled) {
      this.disable();
    }

    this.instance = new ConsoleColor();
  }

  get colors() {
    return this.instance;
  }

  get colorsEnabled() {
    return this.enabled;
  }

  disableColors() {
    this.disable();
    this.initialize();
  }
}

module.exports = new ConsoleColors();
```

Reply:
The suggested code won't work in ESM because in ESM, exports should be statically defined whereas `module.exports = new ConsoleColors();` generates the exports during runtime.

Reply:
> exports should be statically defined whereas module.exports = new ConsoleColors(); generates the exports during runtime.

It will work since we are using the syntax in `commonJS` format, when we will change in ESM format than
we will export like this
```javascript
class ConsoleColors {
  // ...
}

export default new ConsoleColors();
```

then we can import like this 

```javascript
import consoleColors from './ConsoleColors.js';

consoleColors.enable()

```